src/js/views/globals/add-patient/add-patient_views.js

```diff
@@ -0,0 +1,298 @@
+    });
+  },
+  onWatchKeydown(evt) {
+    if (evt.which === ENTER_KEY) {
+      evt.preventDefault();
+      return;
+    }
+  },
+  onWatchChange(text) {
+    const newText = removeNewline(text);
+    this.ui.input.val(trim(newText));
+
+    this.model.set('last_name', newText);
+  },
+});
+
+const ErrorView = View.extend({
+  className: 'add-patient__error',
+  template: hbs`{{ error }}`,
+});
+
+const AddPatientModal = View.extend({
+  className: 'modal add-patient__modal',
+  regions: {
+    firstName: '[data-first-name-region]',
+    lastName: '[data-last-name-region]',
+    dob: '[data-dob-region]',
+    sex: '[data-sex-region]',
+    groups: '[data-groups-region]',
+    save: {
+      el: '[data-save-region]',
+      replaceElement: true,
+    },
+    error: {
+      el: '[data-error-region]',
+      replaceElement: true,
+    },
+  },
+  triggers: {
+    'click .js-close': 'close',
+  },
+  childViewTriggers: {
+    'save': 'save',
+    'cancel': 'close',
+  },
+  modelEvents: {
+    'change': 'showSave',
+  },
+  template: AddPatientTemplate,
+  initialize({ state }) {
+    this.state = state;
+    this.listenTo(state, {
+      'change:error': () => this.showError(state),
+    });
+  },
+  onRender() {
+    this.showFirstNameView();
+    this.showLastNameView();
+    this.showSave();
+    this.showSexDroplist();
+    this.showBirthDatePicker();
+    this.showGroupsComponent();
+  },
+  onClose() {
+    this.destroy();
+  },
+  showFirstNameView() {
+    this.showChildView('firstName', new FirstNameView({
+      model: this.model,
+      state: this.state,
+    }));
+  },
+  showLastNameView() {
+    this.showChildView('lastName', new LastNameView({
+      model: this.model,
+      state: this.state,
+    }));
+  },
+  showSave() {
+    const saveView = new SaveView({
+      model: this.model,
+    });
+
+    this.showChildView('save', saveView);
+  },
+  showSexDroplist() {
+    const sexDroplist = this.showChildView('sex', new SexDroplist());
+
+    this.listenTo(sexDroplist, {
+      'change:selected'(selected) {
+        this.model.set('sex', selected.get('value'));
+      },
+    });
+  },
+  showBirthDatePicker() {
+    this.showChildView('dob', new BirthdateView({
+      model: this.model,
+      state: this.state,
+    }));
+  },
+  showGroupsComponent() {
+    const groupsManager = this.showChildView('groups', new GroupsManagerComponent({
+      member: this.model,
+    }));
+
+    this.listenTo(groupsManager, {
+      'add:member'(patient, group) {
+        const patientGroups = this.model.get('_groups');
+
+        this.model.set('_groups', patientGroups.concat(group.id));
+      },
+      'remove:member'(patient, group) {
+        const patientGroups = this.model.get('_groups');
+
+        this.model.set('_groups', reject(patientGroups, groupId => groupId === group.id));
+      },
+    });
+  },
+  showError() {
+    const error = this.state.get('error');
+
+    if (error === 'Similar patient exists') {
```

Comment:
actually because we're so married to handling only one particular error for the moment, we could ignore the "errorField" altogether and do something like `this. showFirstNameView({ error: true })` and avoid needing to pass state models

Reply:
OK. I had a feeling this would be the case. I think checking the date on validate would be a good idea. In case we end up with a date field that allows users to type out the date in addition to using a picker.